### ISSUE TITLE
Running OLAP SQL with millions of data results duplicate data in result set on 0.7.0

### ISSUE THREAD

**Issue by yingl** [duckdb/duckdb#6470]
### What happens?

We use duckdb to analyze the stock data. When doing analysis, we found randomly it will return more results than expected. So after investigation, we made a constant repro for your debugging.

During repro we found:
1.  if we get the data by "**_duck.read_csv('data.csv')_**" instead, can't repro it.
2. If we roll back the duckdb from 0.7.0 to 0.6.1, can't repro it..
3. Upgrade to 0.7.1.dev407 is fine.

My hardware configuration is 20 cores (intel i7) with 32GB memory.

### To Reproduce

Here are the codes to repro the issue

1. Generate the test data, 4,000,000 rows of data is expected.
```python
import random

with open('data.csv', 'w+') as f:
  f.write('id,code,value\n')
  for i in range(2000):
    for j in range(2000):
      f.write('%d,%d,%d\n' % (i, j, int(random.random() * 100000)))
```
2. Verification code, it will randomly return more than 4,000,000 rows of data.
```python
import duckdb

sql = '''
select id,
       code,
       avg(value) over five as FA,
       avg(value) over ten as FB,
       (lead(value, 1, null) over days) < (lead(value, 1, null) over days) as canBuy,
       (lead(value, 2, null) over days) / (lead(value, 1, null) over days) - 1 as GA1,
       (lead(value, 3, null) over days) / (lead(value, 1, null) over days) - 1 as GA2
from data
window
    five as (partition by code order by id asc rows between 4 preceding and 0 following),
    ten as (partition by code order by id asc rows between 9 preceding and 0 following),
    days as (partition by code order by id asc)
'''

duck = duckdb.connect(database=':memory:', read_only=False)

data = duck.execute('select * from data.csv').df()
expected_rows = data.shape[0]

for i in range(20):
  df = duck.execute(sql).df()
  rows = df.shape[0]
  if rows != expected_rows:
    print('Error in round %d. %d != %d' % (i, rows, expected_rows))
```

### OS:

Ubuntu 20.0.4 X64 on WSL

### DuckDB Version:

0.7.0

### DuckDB Client:

Python

### Full Name:

Lin Ying

### Affiliation:

Dataphi

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mytherin** [comment]
Thanks for the report! This seems like a duplicate of #6272 - which has been fixed in the latest master.

**Comment by Mytherin** [comment]
This should be fixed in v0.7.1
